spec/controllers/members_controller_spec.rb

```diff
@@ -38,4 +38,48 @@
       end
     end
   end
+
+  describe '.create' do
+    let(:member) { Member.create!(name: 'John Doe', email: 'jd@artsymail.com', group_ids: [@group.id]) }
+    before do
+      @group = Group.create!(name: 'Design')
+      @member = member
+    end
+    it 'checks registration email is sent' do
+      expect { post :create, member: { name: 'John Doe', email: 'jd1@artsymail.com', skip_meetings: 0, group_ids: [@group.id] } }.to change(ActionMailer::Base.deliveries, :count).by(1)
+    end
+
+    it 'verifies email and sends welcome email' do
+      expect { get :verify_email, id: @member.email_confirmation_token }.to change(ActionMailer::Base.deliveries, :count).by(1)
+      @member = Member.find_by_email('jd@artsymail.com')
+      expect(@member.email_confirmed).to be true
+    end
+
+    it 'verifies email using invalid token to check member state' do
+      get :verify_email, id: 'random_1212'
+      @member = Member.find_by_email('jd@artsymail.com')
+      expect(@member.email_confirmed).to be false
+    end
+  end
+
+  describe '.edit' do
+    let(:member) { Member.create!(name: 'John Doe', email: 'jd@artsymail.com', group_ids: [@group.id]) }
+
+    before do
+      @group = Group.create!(name: 'Design')
+      @member = member
+    end
+
+    it 'checks member who have not verified email yet does not have access' do
+      get :edit, id: @member
+      expect(response.status).to eq 302
+    end
+  end
+
+  it 'redirects to 404 when a member is not found' do
+    random_member_id = (2 + rand(10)).to_s
```

Comment:
This makes the test unpredictable and does unnecessary work. Once in a  blue moon you'll get the same id as an actual member. Very simply just pass `id: 'invalid'`, or if this needs to be a number `id: -1` for example.